javalib/src/main/scala/java/net/URI.scala

```diff
@@ -135,13 +135,15 @@ final class URI(origStr: String) extends Serializable with Comparable[URI] {
     case _ => false
   }
 
-  def getAuthority(): String = _authority.map(decodeComponent).orNull
+  def getAuthority(): String =
+    _authority.filter(_ != "").map(decodeComponent).orNull
```

Comment:
For example:

```
scala> new java.net.URI("http://foo/a").resolve("///b")
res4: java.net.URI = http://foo/b
```

I'm pretty sure the current implementation would return `http:///b` or even `///b`.